bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java

```diff
@@ -320,10 +336,39 @@ private void updateBattery(MyRenaultHttpSession httpSession) {
                     updateState(CHANNEL_CHARGING_REMAINING_TIME,
                             new QuantityType<Time>(chargingRemainingTime.doubleValue(), MINUTE));
                 }
+                ZonedDateTime batteryStatusUpdated = car.getBatteryStatusUpdated();
+                if (batteryStatusUpdated != null) {
+                    updateState(CHANNEL_BATTERY_STATUS_UPDATED, new DateTimeType(batteryStatusUpdated));
+                }
             } catch (RenaultNotImplementedException e) {
+                logger.warn("Disable cockpit battery update.");
                 car.setDisableBattery(true);
             } catch (RenaultForbiddenException | RenaultUpdateException e) {
+                logger.warn("Error updating battery status.", e);
             }
         }
     }
+
+    private void updateLockStatus(MyRenaultHttpSession httpSession) {
+        if (!car.isDisableLockStatus()) {
+            try {
+                httpSession.getLockStatus(car);
+                updateState(CHANNEL_LOCK_STATUS, new StringType(car.getLockStatus().name()));
+            } catch (RenaultNotImplementedException e) {
+                updateState(CHANNEL_LOCK_STATUS, new StringType(LockStatus.UNKNOWN.name()));
```

Comment:
```suggestion
                updateState(CHANNEL_LOCK_STATUS, UnDefType.UNDEF);
```

Reply:
I will not commit your suggestion with the Web UI due to the signing issues I had to rebase .... But I will adopt it.  The  LockStatus.UNKNOWN is maybe not useful then?

Reply:
I have pushed this change.

Reply:
Sure, also, if you had done that, you wouldn't have noticed compiler error because of missing import. :) I guess if would make sense so set UNDEF also if case of UNKNOWN in line 356. Something like:
```java
LockStatus lockStatus = car.getLockStatus();
updateState(CHANNEL_LOCK_STATUS, localStatus == UNKNOWN ? UnDefType.UNDEF : new StringType(lockStatus.name()));
```

Reply:
@jlaur On reflection keeping the LockStatus.UNKNOWN is helpful if there is a value delivered by the API that I did not expect.  It will give me a clue that I have to debug and add this value mapping.

However I am happy to map UNKNOWN to UNDEF if you think this is better.

Reply:
So perhaps it would then be better to add a **debug** log with this information, but preferably it should also contain the actual value received from the API, unless there is already a **trace** log providing the full response payload?

Reply:
I do a DEBUG log that has the API  request / responses.  Should these be TRACE logs?

Reply:
Yes, added separate comment for that. So I think **UNDEF** would be correct. If a user would report this, trace logs can be enabled, and you should have everything needed for investigating this. Out of curiosity: Can lock status be anything other than locked, unlocked and unknown? Perhaps a read-only switch would be even better then? Some cars even support locking/unlocking through the API.

Reply:
I have changed to API calls to trace.

The API returns either locked or unlocked as a string.  I need an unknown state which is now covered by UNDEF.  So it is technically possible to change the locked status channel to a locked Switch boolean type.  

I do not know if the API will return anything other that the 2 states.  I also do not know why they did not choose a boolean type.  I used a string for flexibility if the API changes.  Also each door status has a opened status... and so a string of "unlocked and 2 doors open" could be added later if it is an information string.  However not sure this is useful in a home automatization context.

I am not sure how users will use the locked status.  I guess a home security rule could check the location and locked status of the car as part of a openHAB home security system rule set.  

That is the design background.  What do you think is better a flexible status string or a Boolean locked?

Reply:
With the current knowledge, a switch type channel seems more appropriate for mapping this into an openHAB context. If you would later find something like "unlocked and 2 doors open", that would result in UNDEF anyway, so changes would need to be made in the binding. In this case it might even make sense to have both a `locked` channel as well as `doorsOpen` or whatever can be derived from such API strings.